auth/store.go

```diff
@@ -164,6 +172,9 @@ type AuthStore interface {
 
 	// AuthInfoFromTLS gets AuthInfo from TLS info of gRPC's context
 	AuthInfoFromTLS(ctx context.Context) *AuthInfo
+
+	// TokenType gets a type of token used by authStore
+	TokenType() int
```

Comment:
If the auth store is only going to support one type of token at a time, it probably shouldn't export this. Probably needs some kind of internal polymorphism/subtyping/interfacing to keep the code paths clean (likely, a jwt.go file) instead of the branching everywhere / having fields in `authStore` that may be irrelevant depending on the configured token type.